pkg/operator/controller/ingress/status.go

```diff
@@ -693,13 +693,28 @@ func computeLoadBalancerStatus(ic *operatorv1.IngressController, service *corev1
 		// Try and find a more specific reason for for the pending status.
 		createFailedReason := "SyncLoadBalancerFailed"
 		failedLoadBalancerEvents := getEventsByReason(operandEvents, "service-controller", createFailedReason)
-		for _, event := range failedLoadBalancerEvents {
-			involved := event.InvolvedObject
-			if involved.Kind == "Service" && involved.Namespace == service.Namespace && involved.Name == service.Name && involved.UID == service.UID {
-				reason = "SyncLoadBalancerFailed"
-				message = fmt.Sprintf("The %s component is reporting SyncLoadBalancerFailed events like: %s\n%s",
-					event.Source.Component, event.Message, "The kube-controller-manager logs may contain more details.")
-				break
+		if len(failedLoadBalancerEvents) > 0 {
+			for _, event := range failedLoadBalancerEvents {
+				involved := event.InvolvedObject
+				if involved.Kind == "Service" && involved.Namespace == service.Namespace && involved.Name == service.Name && involved.UID == service.UID {
+					reason = "SyncLoadBalancerFailed"
+					message = fmt.Sprintf("The %s component is reporting SyncLoadBalancerFailed events like: %s\n%s",
+						event.Source.Component, event.Message, "The kube-controller-manager logs may contain more details.")
+					break
+				}
+			}
+		} else {
+			// The LB service eventually stops reporting events, even if there
+			// are still errors. If the service is still in the pending state
+			// and on the previous update, it was in a failed state, assume it
+			// is still in a failed state even if no events are present.
+			for _, condition := range ic.Status.Conditions {
+				if condition.Type == operatorv1.LoadBalancerReadyIngressConditionType &&
+					condition.Status == operatorv1.ConditionFalse &&
+					condition.Reason == "SyncLoadBalancerFailed" {
+					conditions = append(conditions, condition)
+					return conditions
```

Comment:
Set up the condition and add it later in line 720
```suggestion
					condition.Status == operatorv1.ConditionFalse &&
					condition.Reason == "SyncLoadBalancerFailed" {
					reason = condition.Reason
					message = fmt.Sprintf("No more events occurred after SyncLoadBalancerFailed, and status is still Pending")
					break
```

Reply:
That's a reasonable change in the way this function works, although I'd like to keep the existing message. Even if the event goes away, I think the existing message that lists out some of the messages from previous SyncLoadBalancerFailed events is more informative